portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,30 +24,67 @@
 import java.util.Collections;
 import java.util.List;
 import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that handles comments data.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    private static ArrayList<Comment> comments = new ArrayList<Comment>();
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment");
+
+        // Obtain information form DataStore
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Do not need to do this here. But I think it would probably good practice in future to wrap the datastoreService and create an API that deals only with your applications type of objects.

So you would have a CommentService which can getComments and putComments or the like and this CommentService would encapsulate all the talking to your data storage solution and mapping from its data representation to yours.

Reply:
https://en.wikipedia.org/wiki/Data_access_object

This would make it easy for you to switch your storage layer in the future if you needed to or wanted to as the application grows as all its code would reside in one place